src/vcpkg/commands.ci.cpp

```diff
@@ -434,30 +437,6 @@ namespace vcpkg::Commands::CI
         });
     }
 
-    // This algorithm reduces an action plan to ports that differ from the parent revision, plus dependencies.
-    static void reduce_action_plan(Dependencies::ActionPlan& action_plan, const std::vector<std::string>& parent_hashes)
-    {
-        std::set<PackageSpec> to_keep;
-        for (auto it = action_plan.install_actions.rbegin(); it != action_plan.install_actions.rend(); ++it)
-        {
-            auto abi = it->abi_info.value_or_exit(VCPKG_LINE_INFO).package_abi;
-            auto it_parent = std::find(parent_hashes.begin(), parent_hashes.end(), abi);
-            if (it_parent == parent_hashes.end())
-            {
-                to_keep.insert(it->spec);
-            }
-
-            if (Util::Sets::contains(to_keep, it->spec) && it->plan_type != InstallPlanType::EXCLUDED)
```

Comment:
@ras0219-msft Note that there was an extra condition here: Don't keep specs just for being dependencies of an affected port if the affected port is excluded. I think this can be safely taken to the combined `reduce_actions`, even if it may sometimes reduce the installation plan when no parent hashes are given.